### DISCUSSION TITLE
building clang version error in openeuler(aarch64)

### DISCUSSION THREAD

**Discussion by YutingNie** [apache/doris#26181] [A - General / Q&A]
I'm trying to build doirs on **openeuler os(aarch64)** with doris offical document
![image](https://github.com/apache/doris/assets/104416402/d0d9f123-fb44-4c75-9770-0721e7025cd7)
However, the following error occurred during compilation:
```
CMake Error at CMakeLists.txt:269 (message):
Need Clang version at least 16.
```
![image](https://github.com/apache/doris/assets/104416402/fbfff5cb-bc1d-4146-a489-c5a7efc7c6ad)

According to the Doris documentation, the downloaded ldb_toolchain version is 0.9, and the included Clang version is 13. 
![image](https://github.com/apache/doris/assets/104416402/1d28996c-e203-40c4-a296-5a3fd1845e6d)
This means that ldb_toolchain version 0.9 unable to building doirs on openeuler os(aarch64). 
However, I have searched for information and found ldb_toolchain version 0.17, which includes Clang 16. Unfortunately, this version does not have an sh file for the aarch64 architecture. Could you please advise on how to resolve this issue?
Here is my system info:
![image](https://github.com/apache/doris/assets/104416402/bd46b05b-7861-4b36-9fd4-9d8aa77287f2)

**Comment by huayucai** [comment]
also met the problem . In file  'env.sh'  I made this change:
![image](https://github.com/apache/doris/assets/135694301/dd2f9f60-8385-4b02-a279-bdd52f6b1408)
by using gcc to avoid using clang. you can have a try

**Reply by YutingNie** [reply]
thanks for you reply
I finally change the CMakeList.txt to solve this problem.
![image](https://github.com/apache/doris/assets/104416402/8674feef-cc4d-45e6-99ae-366ba8461f57)
But a new issue has arisen, compiling Doris shows "Apache Hadoop Common BUILD FAILURE," 
while I can successfully compile it separately in the Hadoop Common directory. Have you encountered this issue before?
![image](https://github.com/apache/doris/assets/104416402/95ccb207-614a-4880-919d-2ae921ee02b0)
